sdk/trace/trace_test.go

```diff
@@ -217,13 +217,26 @@ func TestSetName(t *testing.T) {
 	}
 }
 
-func TestRecordingIsOn(t *testing.T) {
-	tp := NewTracerProvider()
-	_, span := tp.Tracer("Recording on").Start(context.Background(), "StartSpan")
-	defer span.End()
-	if span.IsRecording() == false {
-		t.Error("new span is not recording events")
-	}
+func TestSpanIsRecording(t *testing.T) {
+	t.Run("is recording on", func(t *testing.T) {
+		tp := NewTracerProvider()
+		_, span := tp.Tracer("Recording on").Start(context.Background(), "StartSpan")
+		defer span.End()
```

Comment:
Rather than `defer`ing the `span.End()` call, perhaps end the span and then assert that it is no longer recording.